src/app/problems/problem-create-edit/java-create-edit-snippet/java-create-edit-snippet.component.html

```diff
@@ -31,7 +31,8 @@ <h2 class="tui-form__header">Submitted Files</h2>
     </div>
     <div class="tui-form__row">
         <h2 class="tui-form__header">Variables</h2>
-        <p class="tui-form__field-note">
+        <app-variation-types-selector [variationControl]="variationControl"></app-variation-types-selector>
```

Comment:
see how app `app-variables-editor` work with the formControlName instead of the formControl itself. It might simplify things a lot if you do the same.